docs/appendices/release-notes/unreleased.rst

```diff
@@ -51,6 +51,9 @@ Breaking Changes
 Changes
 =======
 
+- Changed the error code for the psql protocol when a column alias is
+  ambiguous from `XX000` `internal_error` to `42P09` `ambiguous_alias`.
```

Comment:
Maybe we could already start grouping these changes into one entry or so. But doing that with the next PR is also fine.